enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -1230,6 +1234,9 @@ func (s *SchedulerServer) unclaimTask(ctx context.Context, taskID, leaseID, reco
 		return status.NotFoundErrorf("unable to release task claim for task %s", taskID)
 	}
 	log.CtxDebugf(ctx, "Released task claim in Redis (reconnecting=%t)", reconnectToken != "")
+
+	action_merger.DeletePendingExecution(ctx, s.rdb, taskID)
```

Comment:
If a task is unclaimed, it doesn't mean it's done. It can still be retried on a different executor. Do we need to remove the link here?

Reply:
Yeah, I think that makes sense. I had it in there because I was worried the task could be unclaimed and not reclaimed, and then get stuck again, but with only a 20 second blackhole period I think it's fine to let the entry sit around on unclaim.